File: src/main/java/giraffe/miausicbox/model/Novelty.java
package giraffe.miausicbox.model;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;

import com.fasterxml.jackson.annotation.JsonView;

import giraffe.miausicbox.user.User;

@Entity
public class Novelty {

	public interface Basic extends User.Basic, Band.Basic {}
	
	// Attributes
	@JsonView(Basic.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(Basic.class)
	@OneToOne
	private User user;
	
	@JsonView(Basic.class)
	@OneToOne
	private Band band;
	
	@JsonView(Basic.class)
	private String date;
	
	@JsonView(Basic.class)
	private Boolean joined;
	
	// Constructor
	public Novelty() {}
	
	public Novelty(
			User user,
			Band band,
			String date,
			Boolean joined
			) {
		super();
		this.user= user;
		this.band = band;
		this.date = date;
		this.joined = joined;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Band getBand() {
		return band;
	}

	public void setBand(Band band) {
		this.band = band;
	}

	public String getDate() {
		return date;
	}

	public void setDate(String date) {
		this.date = date;
	}

	public Boolean isJoined() {
		return joined;
	}

	public void setJoined(Boolean joined) {
		this.joined = joined;
	}
	
	// ToString
	@Override
	public String toString() {
		String str = this.getUser().getUserName();
		if (this.joined) {
			str += " joined ";
		} else {
			str += " left ";
		}
		return (str + this.getBand().getGroupName());
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Novelty other = (Novelty) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getUser(), other.getUser())
				&& (Objects.equals(this.getBand(), other.getBand()))
				&& (Objects.equals(this.isJoined(), other.isJoined()))
				&& (Objects.equals(this.getDate(), other.getDate())));
	}
	
}

File: src/main/resources/static/app/classes/Novelty.ts
/**
 * Class that represent the information related to MiausicBox novelties
 * @class Message
 */
import {Band} from "./Band";
import {User} from "./User";

export class Novelty{

  /* Attributes */
  private _id: number;
  private _user: User;
  private _band: Band;
  private _date: Date;
  private _joined: boolean

  /* Constructor */
  constructor(id:number, user:User, band: Band, date:Date, joined: boolean) {
    this._id = id;
    this._user = user;
    this._band = band;
    this._date = date;
    this._joined = joined;
  }

  /* Getters & Setters */
  get id():number {
    return this._id;
  }

  get user():User {
    return this._user;
  }

  get band():Band {
    return this._band;
  }

  get date():Date {
    return this._date;
  }

  get joined():boolean {
    return this._joined;
  }

  set id(id:number) {
    this._id = id;
  }

  set user(user:User) {
    this._user = user;
  }

  set band(band:Band) {
    this._band =  band;
  }

  set date(date:Date) {
    this._date = date;
  }

  set joined(joined:boolean) {
    this._joined = joined;
  }

  /** Return if a Novelty is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any) {
    if (!(object instanceof Novelty)) {
      return false;
    } else {
      var novelty:Novelty = object;
      return (novelty.user.equals(this.user) &&
              novelty.band.equals(this.band) &&
              novelty.joined == this.joined);
    }
  }

}
